src/main/java/uk/ac/ebi/intact/service/complex/ws/writer/ComplexWriterFactory.java

```diff
@@ -0,0 +1,66 @@
+package uk.ac.ebi.intact.service.complex.ws.writer;
+
+import psidev.psi.mi.jami.commons.MIWriterOptionFactory;
+import psidev.psi.mi.jami.datasource.InteractionWriter;
+import psidev.psi.mi.jami.factory.InteractionWriterFactory;
+import psidev.psi.mi.jami.json.InteractionViewerJson;
+import psidev.psi.mi.jami.json.MIJsonOptionFactory;
+import psidev.psi.mi.jami.json.MIJsonType;
+import psidev.psi.mi.jami.model.ComplexType;
+import psidev.psi.mi.jami.model.InteractionCategory;
+import psidev.psi.mi.jami.xml.PsiXmlVersion;
+import uk.ac.ebi.intact.dataexchange.psimi.xml.IntactPsiXml;
+import uk.ac.ebi.intact.service.complex.ws.ComplexExportFormat;
+
+import java.io.StringWriter;
+import java.util.Map;
+
+public class ComplexWriterFactory {
```

Comment:
We have 2 types of writers.
- `ComplexWriter` and classes that implement it: writers that write complexes into a string
- `SerialisedComplexesWriter` and classes that implement it: writers that add serialised complexes strings into the right format, with the right header and footer just once.